### PR TITLE
feat: add simple chat example using MLX (Apple Silicon ML framework)

### PR DESCRIPTION
**Pull request by fblissjr**
## 📝 Summary

adds a simple chat example pattern that can be extended using local models using Apple's MLX framework (Apple Silicon only)

## 🔍 Description of Changes
`mlx_chat.py` added to `examples/ai/chat`

context:
[`mlx_lm`](https://github.com/ml-explore/mlx-examples/tree/main/llm) is a library for running large language models on Apple Silicon.
[`mlx`](https://github.com/ml-explore/mlx) is a framework for running machine learning models on Apple Silicon.

## 📋 Checklist

- [X] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [X] I have run the code and verified that it works as expected.

I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

**Comment by fblissjr** [comment]
> Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://marimo.io/cla) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
> 
> I have read the CLA Document and I hereby sign the CLA
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request. Posted by the **CLA Assistant Lite bot**.

I have read the CLA Document and I hereby sign the CLA

**Comment by fblissjr** [comment]
I have read the CLA Document and I hereby sign the CLA

File: examples/ai/chat/mlx_chat.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    load_model_button = mo.ui.run_button(label="Load Model")
+    
+    mo.hstack([MODEL_ID, load_model_button])
+    return (MODEL_ID, load_model_button,)
+
+@app.cell
+def __(MODEL_ID, get_model_path, load, load_model_button, mo):
+    mo.stop(not load_model_button.value, "Click 'Load Model' to proceed")
+    
+    try:
```
Comment (mscolnick) [inline examples/ai/chat/mlx_chat.py:98]: you could use `mo.status.spinner()`: https://docs.marimo.io/api/status.html#spinner
and update the titles to each message, 

or an easier refactor, you could do `mo.output.append(mo.md(..` so you can track progress

File: examples/ai/chat/mlx_chat.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            except Exception as e:
+                raise ValueError(f"Error downloading model from Hugging Face: {str(e)}")
+
+    return (get_model_path,)
+
+@app.cell
+def __(mo):
+    MODEL_ID = mo.ui.text(
+        label="Hugging Face Model Repo or Local Path", 
+        value="mlx-community/Llama-3.2-3B-Instruct-bf16",
```
Comment (mscolnick) [inline examples/ai/chat/mlx_chat.py:84]: just curious, is `mlx-community/Llama-3.2-3B-Instruct-bf16` the smallest model? i am finding it take a long time to download

**Review by akshayka — COMMENTED** [review]
This is awesome!

Small request: can you add a bullet point to `chat/README.md` describing what this example is? just one sentence

File: examples/ai/chat/mlx_chat.py (hunk 2 above)
Comment (fblissjr) [reply examples/ai/chat/mlx_chat.py:84]: No there are smaller ones, this is just the  full precision one which is most accurate. You can get 8 bit or 4 bit quant, or also just get the smaller llama 3.2 1B parameter model. 

Here’s search results for all the post-converted MLX community llama 3.2 models. Note that instruct is the ‘chat’ one. Without it is the base model, which is autocomplete or finetuning your own use case. Base models are my favorite, and planning to do a “notebook” mode example next. 

https://huggingface.co/models?search=mlx%20llama%203.2

**Review by fblissjr — COMMENTED** [review]
Done!

Excited to finally contribute something after all this time using it. 😂

File: examples/ai/chat/mlx_chat.py (hunk 1 above)
Comment (fblissjr) [reply examples/ai/chat/mlx_chat.py:98]: This looks much more elegant. I hit a time limit with this today, but will circle back in the morning. Thanks for the tip.

**Pull request merged**